pkg/cloud/services/autoscaling/autoscalinggroup.go

```diff
@@ -156,6 +157,7 @@ func (s *Service) CreateASG(scope *scope.MachinePoolScope) (*expinfrav1.AutoScal
 		DefaultCoolDown:      scope.AWSMachinePool.Spec.DefaultCoolDown,
 		CapacityRebalance:    scope.AWSMachinePool.Spec.CapacityRebalance,
 		MixedInstancesPolicy: scope.AWSMachinePool.Spec.MixedInstancesPolicy,
+		AvailabilityZones:    scope.AWSMachinePool.Spec.AvailabilityZones,
```

Comment:
Is there any additional implementation needed other than adding it to the input? Seems sane to me.

Reply:
I believe that should do the trick. I am just not sure about additional steps since `AutoscalingGroup` is an api tye. I'll try to figure it out.

Reply:
It might be easiest to create a PR upstream and see which tests fail :P

Reply:
True :D